function-aws-api-proxy/src/main/java/io/micronaut/function/aws/proxy/MicronautResponseWriter.java

```diff
@@ -112,8 +115,20 @@ public AwsProxyResponse writeResponse(
                     status + " " +
                             Response.Status.fromStatusCode(status.getCode()).getReasonPhrase());
         }
-
+        awsProxyResponse.setHeaders(getHeaders(awsProxyResponse));
         Timer.stop(TIMER_NAME);
         return awsProxyResponse;
     }
+
+    private static Map<String, String> getHeaders(AwsProxyResponse awsProxyResponse) {
+        Headers multiValueHeaders = awsProxyResponse.getMultiValueHeaders();
+        Map<String, String> headers = new HashMap<>();
```

Comment:
use a linked hash map to retain order

Reply:
resolved via https://github.com/micronaut-projects/micronaut-aws/pull/1338/commits/0f624cb9d920c96333901b5bdc2c1dcd1663787c